pkg/distributor/distributor_test.go

```diff
@@ -3308,6 +3310,14 @@ func TestDistributor_LabelNames(t *testing.T) {
 			expectedResult:    []string{labels.MetricName, "reason", "status"},
 			expectedIngesters: numIngesters,
 		},
+		"should filter metrics by single matcher and apply limit": {
```

Comment:
Could you please add some test cases to cover the following scenarios:
* hints are provided but limit is set to 0
* limit is equal to the number of items that could be returned
* limit is greater than the number of items that could be returned

(same applies for `LabelValuesForLabelName` tests)